synapse/handlers/relations.py

```diff
@@ -534,27 +535,16 @@ async def get_bundled_aggregations(
                     "chunk": annotations
                 }
 
-        # Fetch other relations per event.
-        for event in events_by_id.values():
-            # Fetch any references to bundle with this event.
-            references, next_token = await self.get_relations_for_event(
-                event.event_id,
-                event,
-                event.room_id,
-                RelationTypes.REFERENCE,
-                ignored_users=ignored_users,
-            )
+        # Fetch any references to bundle with this event.
+        references_by_event_id = await self.get_references_for_events(
+            events_by_id.keys(), ignored_users=ignored_users
+        )
+        for event_id, references in references_by_event_id.items():
             if references:
-                aggregations = results.setdefault(event.event_id, BundledAggregations())
-                aggregations.references = {
+                results.setdefault(event_id, BundledAggregations()).references = {
                     "chunk": [{"event_id": ev.event_id} for ev in references]
                 }
 
-                if next_token:
-                    aggregations.references["next_batch"] = await next_token.to_string(
-                        self._main_store
-                    )
```

Comment:
This is an unspecced field, see #14486.

Reply:
Err, no. Not at all. Oops. 😢

Reply:
See https://matrix.to/#/!XXSJTvRPInupfUgQVb:matrix.org/$8Yo_XpJrmjgXzZwn4Xh8hZiK9SkFwYzStTr_3O3eMmc?via=pixie.town&via=matrix.org&via=element.io I've also poked some people internally.